Tensile/Source/client/source/HardwareMonitor.cpp

```diff
@@ -386,9 +390,178 @@ namespace Tensile
                 else
                     m_fanValues[i] += newValue;
             }
+            
+            rsmi_gpu_metrics_t gpuMetrics;
+            auto status = rsmi_dev_gpu_metrics_info_get(m_smiDeviceIndex, &gpuMetrics);
+            if(status != RSMI_STATUS_SUCCESS)
+            {
+                m_freqValues[0]        = std::numeric_limits<uint16_t>::max();
+                m_powerValues[0]       = std::numeric_limits<uint16_t>::max();
+                m_temperatureValues[0] = std::numeric_limits<uint16_t>::max();
+            }
+            else
+            {
+                m_freqValues.push_back(gpuMetrics.average_gfxclk_frequency);
+                m_powerValues.push_back(gpuMetrics.average_socket_power);
+                m_temperatureValues.push_back(gpuMetrics.temperature_hotspot);
+            }
 
             m_dataPoints++;
         }
+        
+        double HardwareMonitor::getMedianGfxFrequency()
+        {
+            assertNotActive();
+            if(m_dataPoints == 0)
+                throw std::runtime_error("No data points collected!");
+                
+            if (m_freqValues.size() != 0)
+            {
+                if(m_freqValues[0] == std::numeric_limits<uint16_t>::max())
+                    return std::numeric_limits<double>::quiet_NaN();
+
+                 size_t FrequencyMidValueIndex = m_freqValues.size() / 2;
+                 std::nth_element(m_freqValues.begin(), m_freqValues.begin()+FrequencyMidValueIndex, m_freqValues.end());
+                 return static_cast<double> (m_freqValues[FrequencyMidValueIndex]);
+            }
+            
+            throw std::runtime_error("Can't read Gfx value that wasn't requested: ");
+        }
+        
+        double HardwareMonitor::getAverageGfxFrequency()
```

Comment:
I cannot find a call to this function. Are we using this? Anyhow, it is ok to keep it.

Reply:
These functions are place holder for future enhancement.  I am going to add tensile switch , if it is enabled I will invoke these functions  getAverageGfxFrequency(), logMinMaxMedianAverage(). 

logMinMaxMedianAverage output will be something like this for each GEMM kernel execution.
ROCm SMI API consolidated frequency,power,temperature data

GFX Value	PPT0_value	Temperature
1245         	253		        36
1245         	108		        36
...
				  Min    Max	      Average	Median
GFX Frequency	  898     1257             1213.17   1256    
Power Value		  108     262                237.733   262     
Temperature		  36      38                   36.7          37